src/server/scripts/Northrend/Naxxramas/boss_razuvious.cpp

```diff
@@ -113,33 +134,111 @@ class boss_razuvious : public CreatureScript
             {
                 switch (eventId)
                 {
+                    case EVENT_ATTACK:
+                        SetCombatMovement(true);
+                        if (me->GetVictim())
+                            me->GetMotionMaster()->MoveChase(me->EnsureVictim());
+                        break;
                     case EVENT_STRIKE:
                         DoCastVictim(SPELL_UNBALANCING_STRIKE);
-                        events.ScheduleEvent(EVENT_STRIKE, 30000);
+                        events.ScheduleEvent(EVENT_STRIKE, 6 * IN_MILLISECONDS);
                         return;
                     case EVENT_SHOUT:
                         DoCastAOE(SPELL_DISRUPTING_SHOUT);
-                        events.ScheduleEvent(EVENT_SHOUT, 25000);
+                        events.ScheduleEvent(EVENT_SHOUT, 16 * IN_MILLISECONDS);
                         return;
                     case EVENT_KNIFE:
                         if (Unit* target = SelectTarget(SELECT_TARGET_RANDOM, 0, 45.0f))
                             DoCast(target, SPELL_JAGGED_KNIFE);
-                        events.ScheduleEvent(EVENT_KNIFE, 10000);
-                        return;
-                    case EVENT_COMMAND:
-                        DoPlaySoundToSet(me, SOUND_COMMND);
-                        events.ScheduleEvent(EVENT_COMMAND, 40000);
+                        events.ScheduleEvent(EVENT_KNIFE, urandms(10,15));
                         return;
                 }
             }
 
             DoMeleeAttackIfReady();
         }
+
+        void Reset() override
+        {
+            SetCombatMovement(false);
+            _Reset();
+        }
     };
 
 };
 
+class npc_dk_understudy : public CreatureScript
+{
+    public:
+        npc_dk_understudy() : CreatureScript("npc_dk_understudy") { }
+
+        struct npc_dk_understudyAI : public ScriptedAI
+        {
+            npc_dk_understudyAI(Creature* creature) : ScriptedAI(creature), _instance(creature->GetInstanceScript()), bloodStrikeTimer(0) { }
+
+            void EnterCombat(Unit* /*who*/) override
+            {
+                me->SetUInt32Value(UNIT_NPC_EMOTESTATE, EMOTE_ONESHOT_NONE);
+                if (Creature* razuvious = ObjectAccessor::GetCreature(*me, _instance->GetGuidData(DATA_RAZUVIOUS)))
+                    razuvious->AI()->DoZoneInCombat(nullptr, 250.0f);
+            }
+
+            void JustReachedHome() override
+            {
+                if (_instance->GetBossState(BOSS_RAZUVIOUS) == DONE)
+                    me->DespawnOrUnsummon();
+                else
+                    ScriptedAI::JustReachedHome();
+            }
+
+            void UpdateAI(uint32 diff) override
+            {
+                if (!me->isPossessedByPlayer() && !UpdateVictim())
+                    return;
+
+                if (!me->isPossessedByPlayer())
```

Comment:
The only charm mechanic it can be affected by is possess (the orb and priest MC)